### DISCUSSION TITLE
Question about adding mask orders image.

### DISCUSSION THREAD

**Discussion by phamkhactu** [kornia/kornia#2651] [Questions] [answered]
Thanks for great working.

I tried stitch images with order images. As the [question in stackoverflow](https://stackoverflow.com/questions/70897908/how-do-i-format-the-mask-parameter-for-cv2-detail-bestof2nearestmatcher-apply2-f). If I have  a mask matrix  for Apply2 function, function auto works with orders. With Kornia, how can I pass mask matrix for stitching.

you can provide some code example if you have feature like that, if not, I think it maybe is a feature request for making Koria package more useful.
Thank you very much!

**Comment by edgarriba** [comment, marked as answer]
Have you tried zeroing the image based on the mask (hadamard prod) ? /cc @ducha-aiki

**Comment by phamkhactu** [comment]
> Have you tried zeroing the image based on the mask

Yes, I have. I run it successfully with order images.

**Comment by edgarriba** [comment]
@phamkhactu does it mean that it solved your problem ?

**Comment by phamkhactu** [comment]
> @phamkhactu does it mean that it solved your problem ?

yes, it does, it helps to reduce time for matching picture with picture. That is why I  require feature.

**Comment by edgarriba** [comment]
if i understand correctly you are asking for a `mask` parameter that internally applies elementwise product to the input images ? If that's the case, i'm not quite sure we might go for it since it's a pre-step that the user can always apply.

**Comment by phamkhactu** [comment]
> if i understand correctly you are asking for a `mask` parameter that internally applies elementwise product to the input images ? If that's the case, i'm not quite sure we might go for it since it's a pre-step that the user can always apply.

Yes, IF I can give mask, maybe algorithm not need calculate relative images. Some time, calculate not right. So can give it in function becomes better.

**Comment by edgarriba** [comment]
i don't the model architecture is designed for such scenario. Seems more like a research project to fine tune the network (which we don't provide the code) and input RGB+mask to learn more accurate attention maps

**Comment by phamkhactu** [comment]
> i don't the model architecture is designed for such scenario. Seems more like a research project to fine tune the network (which we don't provide the code) and input RGB+mask to learn more accurate attention maps

Yeah, 
Thank you for more your details information.
